libs/core/include/ScaleFactorsManager.hpp

```diff
@@ -21,16 +21,16 @@ struct MuonIso;
 class ScaleFactorsManager {
  public:
   static ScaleFactorsManager &GetInstance() {
-    static ScaleFactorsManager instance;
-    return instance;
+    static ScaleFactorsManager* instance = new ScaleFactorsManager();
+    return *instance;
   }
 
   ScaleFactorsManager(ScaleFactorsManager const &) = delete;
   void operator=(ScaleFactorsManager const &) = delete;
 
   std::map<std::string, float> GetPUJetIDScaleFactors(std::string name, float eta, float pt);
   std::map<std::string, float> GetMuonScaleFactors(std::string name, float eta, float pt);
-  std::map<std::string, float> GetDSAMuonScaleFactors(std::string name, float eta, float pt);
+  std::map<std::string, float> GetDSAMuonScaleFactors(std::string patname, std::string dsaname, float eta, float pt);
```

Comment:
Could you please change it to `patName` and `dsaName`, to follow the convention?